src/ngraph/op/concat.cpp

```diff
@@ -142,3 +143,61 @@ void op::Concat::generate_adjoints(autodiff::Adjoints& adjoints, const OutputVec
         pos = next_pos;
     }
 }
+
+namespace
+{
+    template <element::Type_t ET>
+    inline bool
+        evaluate(const HostTensorVector& args, const HostTensorPtr& out, int64_t concatenation_axis)
+    {
```

Comment:
You need to figure the output shape from the input host tensors and set it on the output, in case some of the inputs are dynamic.

Reply:
shape inference should have handled for static, wasn't sure we are supporting dynamic inputs? will update.

Reply:
We are supporting dynamic.

Reply:
https://github.com/NervanaSystems/ngraph/pull/4705/commits